ddl/index.go

```diff
@@ -680,29 +670,32 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 	return ver, errors.Trace(err)
 }
 
-// checkAndInitLightningBackfillCtx determines which backfill process will be used. The following three processes are available:
-//  1. Txn backfill(original): all the index KVs are written through the transaction interface.
-//  2. Lightning backfill: the index KVs are encoded to SST files and imported to the storage directly.
-//     The incremental index KVs written by DML are redirected to a temporary index.
-//     After the backfill is finished, the temporary index records are merged back to the original index.
-//  3. Txn-merge backfill: the backfill index KVs are written through the transaction interface.
-//     The incremental index KVs written by DML are redirected to a temporary index.
-//     After the backfill is finished, the temporary index records are merged back to the original index.
-func checkAndInitLightningBackfillCtx(ctx context.Context, job *model.Job, indexInfo *model.IndexInfo) {
+// pickBackfillProcess determines which backfill process will be used.
+func pickBackfillProcess(w *worker, job *model.Job, indexInfo *model.IndexInfo) model.ReorgType {
```

Comment:
When crash and restart job,  seems there is no this part logic in this pr. you will do it later?